pep-9999.rst

```diff
@@ -0,0 +1,193 @@
+PEP: 9999
+Title: Recording provenance of installed packages
+Author: Fridolín Pokorný <fridolin.pokorny at gmail.com>
+Sponsor: Donald Stufft <donald@stufft.io>
+PEP-Delegate:
+Discussions-To: https://discuss.python.org/t/pep-705-recording-provenance-of-installed-packages/23340
+Status: Draft
+Type: Process
+Content-Type: text/x-rst
+Created: 30-Jan-2002
+Post-History:
+
+Abstract
+========
+
+This PEP describes a way to record provenance of Python distributions
+installed.  The record is created by an installer and is available to users in
+a form of a JSON file ``provenance_url.json`` in ``.dist-info`` directory. The
+mentioned JSON file captures additional metadata to allow recording URL to a
+Python distribution together with the installed Python distribution hash. This
+proposal is built on top of :pep:`610` following `its corresponding canonical
+PyPA spec
+<https://packaging.python.org/en/latest/specifications/direct-url/>`__ and
+complements ``direct_url.json`` with ``provenance_url.json`` file when packages
+are identified by a name, and either a version
+
+Motivation
+==========
+
+Installing a Python package involves downloading a distribution from an index
+and extracting its content to an appropriate place. After the installation
+process is done, information about the distribution used as well as its source
+is generally lost. Nevertheless, there are use cases for keeping records of
+distributions used for installing packages and their provenance.
+
+Python wheels can be built with different compiler flags or supporting
+different wheel tags.  In both cases, users might get into a situation in which
+multiple wheels might be considered by installers (possibly from different
+package indexes) and immediately finding out which wheel file was actually used
+during the installation might be helpful. This way, tools reporting software
+installed, such as tools reporting SBOM (Software Bill of Material), might give
+more accurate reports.
+
+The motivation described in this PEP is an extension to :pep:`610`.  Besides
+stating information about packages installed from an URL, installers SHOULD
+record information also for packages installed from Python package indexes when
+identified by their name, and optionally their version.
+
+Specification
+=============
+
+The ``provenance_url.json`` file SHOULD be created in the ``*.dist-info``
```

Comment:
Just to be explicit here - an exception might be legacy installations. Do they need to be explicitly stated (there is SHOULD anyway)?